### PR TITLE
Expose is_restart_required property on files in the search/detail API

### LINKED ISSUE #5772: Add is_restart_required property to File api

### PR DESCRIPTION
**Pull request by diox**
Fix #5772 

Will require a reindex.

### CODE REVIEW

**Comment by diox** [comment]
I'm going to take a shot at unifying this mess into this PR, I think it's the perfect opportunity.

**Comment by diox** [comment]
@EnTeQuAk re-r? with the field name change? I moved everything I could to "is_restart_required". Removed some bits in the legacy disco pane because I could not find any matching css/js, and I don't think we care that much about the legacy disco at this point anyway.

File: src/olympia/files/models.py [hunk 1]
```diff
@@ -63,6 +63,9 @@ class File(OnChangeMixin, ModelBase):
     status = models.PositiveSmallIntegerField(
         choices=STATUS_CHOICES.items(), default=amo.STATUS_AWAITING_REVIEW)
     datestatuschanged = models.DateTimeField(null=True, auto_now_add=True)
+    is_restart_required = models.BooleanField(default=False)
+    # FIXME: remove no_restart once is_restart_required is deployed to
+    # production.
```
Comment (EnTeQuAk) [inline src/olympia/files/models.py]: 👍 that's perfect. Can we have and link to a follow-up github issue?

File: src/olympia/migrations/954-add-file-is-restart-required.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- is_restart_required replaces the old no_restart field. It's the exact opposite.
+ALTER TABLE `files`
+    ADD COLUMN `is_restart_required` bool NOT NULL,
+    MODIFY `no_restart` bool;  -- Allow nulls to prepare for column removal next push.
+UPDATE `files` SET `is_restart_required` = NOT `no_restart`;
```
Comment (EnTeQuAk) [inline src/olympia/migrations/954-add-file-is-restart-required.sql]: I wonder if ops is going to kill us for this. Might be worth giving them a heads-up and check timing on dev and stage.

File: static/js/zamboni/addon_details.js [hunk 3]
```diff
@@ -25,11 +25,4 @@ $(document).ready(function() {
 
     var etiquette_box = $("#addons-display-review-etiquette").hide();
     $("#short-review").focus(function() { etiquette_box.show("fast"); } );
-
-    /* Restart required box. (Only shown in Fx4+). */
-    var requires_restart = $('#addon-summary #requires-restart');
-    if (requires_restart.length && z.browser.firefox
-        && VersionCompare.compareVersions(z.browserVersion, '4.0a1') > 0) {
-        requires_restart.show();
-    }
```
Comment (EnTeQuAk) [inline static/js/zamboni/addon_details.js]: 🎉 🌮 🍻

File: src/olympia/files/models.py (hunk 1 above)
Comment (diox) [inline src/olympia/files/models.py]: I created https://github.com/mozilla/addons-server/issues/5881 ; I removed the field entirely in order to prepare for the removal of the column.

File: src/olympia/migrations/954-add-file-is-restart-required.sql (hunk 2 above)
Comment (diox) [inline src/olympia/migrations/954-add-file-is-restart-required.sql]: Yep. I'm reasonably confident: I don't know about prod, but dev and stage have less than 170k rows each, and going through the table (`SELECT * FROM files ORDER BY RAND() LIMIT 10`) is pretty quick.

Comment (EnTeQuAk) [inline src/olympia/migrations/954-add-file-is-restart-required.sql]: perfect, that sounds more than reasonable.

**Pull request merged**